presto-main/src/main/java/com/facebook/presto/cost/FilterStatsCalculator.java

```diff
@@ -135,18 +135,19 @@ public PlanNodeStatsEstimate filterStats(
     public PlanNodeStatsEstimate filterStats(
             PlanNodeStatsEstimate statsEstimate,
             RowExpression predicate,
-            ConnectorSession session)
+            ConnectorSession session,
+            Optional<Session> systemSession)
```

Comment:
when would the session not be present? My main concern would be if there are cases where the config won't be checked.

Reply:
There was a discussion in the original PR on this: https://github.com/prestodb/presto/pull/21236#discussion_r1434243541

Mainly an issue due to the implementation of `ConnectorFilterStatsCalculatorService`. Most of the time when we call `filterStats, we will have a `Session` object, but when the stats filtering is called from `ConnectorFilterStatsCalculatorService`, we only get a `ConnectorSession`. ConnectorSession and Session are two separate classes with different methods, and you can't check for a system session property through a connector session. 

There is a piece of coded added here which is where the session is populated: https://github.com/prestodb/presto/blob/18d2e7aab4ae020e9abf380ac87f8c08225ab625/presto-main/src/main/java/com/facebook/presto/cost/ConnectorFilterStatsCalculatorService.java#L60-L62

It should usually be there as there are only two implementations of ConnectorSession (`FullConnectorSession` and `TestingConnectorSession`), but I didn't want to re-design the session APIs in the histogram PR for this, so I introduced the "hack" here to get the underlying session instance

Reply:
yeah, the whole connector session abstraction is kind of weird.  How much value do you think we get from being able to set this on a per-session basis.  because a nicer alternative for this problem is to only have a configuration property for this and inject it directly into filterStatsCalculator, Then we don't have to worry about passing the session around or expecting the session to probably be there even though the interface doesn't guarantee it.

Reply:
Personally, I think it's been pretty useful during some of my manual testing to be able to set the session property within the same query runner/cluster and not have to change config+restart. Additionally, with our current infra it's a little bit of a headache to set the file-based configs flags. Session properties are nice because it doesn't require us to restart the whole cluster when I just want to see how a histogram affects a query plan's estimates and shape.

I could take a stab at shuffling some fields to see if we could get the underlying Session from the ConnectorsSession instance, but I'd probably punt it to another PR

Reply:
can you add a comment then in the meantime explaining why it's optional and the plan to fix it.

Reply:
Comment added